### PR TITLE
Mark notification as read without visiting discussion

### LINKED ISSUE #151: Marking notifications as read without visiting the discussion
Sometimes I just want to approve that I've seen a notification without actually visiting the referenced post. e.g. when Toby goes on a liking spree ;)

Could we have a button next to each notification (or each group, not sure) similar to the one that marks all notifications as read?

### PR DESCRIPTION
**Pull request by datitisev**
**Fixes #151**

**Changes proposed in this pull request:**
- `Notification` component now has a checkbox to mark notification as read

**Reviewers should focus on:**
- Do we really need `slidable` with this? The current structure of notifications made it hard, and I was unable to integrate the `slidable` util... though I have never used it before, so that may be why.
- Is `core.forum.notifications.mark_as_read_tooltip` ok?

**Screenshot**
screen recording 2018-10-06 at 10 01 40 am

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `php vendor/bin/phpunit`).

**Required changes:**

- [ ] Related core extension PRs: `flarum/english` needed, waiting on confirmation for translation key

### CODE REVIEW

**Comment by matteocontrini** [comment]
Wouldn't it be better to show the tick only when hovering the notification?

**Comment by datitisev** [comment]
@matteocontrini Hm, seems like I forgot a crucial part of _"Style this button to float right, only show on hover, and be a bit smaller than normal."_. I'll fix that when I have time, thanks 😂.

**Comment by franzliedke** [comment]
No transition necessary, as far as I am concerned. It's just distracting.

**Comment by franzliedke** [comment]
I'll leave this for @tobscure to merge, and decide about the `slidable` thing.

**Comment by datitisev** [comment]
@franzliedke I tried to implement the `slidable` util, but there weren't enough use cases for me to understand how it worked. It all seemed very confusing. I attempted to make it work, but weird stuff ended up happening 🤔 

> Do we really need slidable with this? The current structure of notifications made it hard, and I was unable to integrate the slidable util... though I have never used it before, so that may be why.

**Comment by franzliedke** [comment]
Ah sorry, you already had explained. Missed that part. I'm fine without it, leaving for Toby, though.

**Comment by luceos** [comment]
@datitisev squash merge when green please ;)

**Comment by datitisev** [comment]
@luceos what do you mean when green ? tests ? also why is this still out of date lol ⚰️

File: less/forum/NotificationList.less [hunk 1]
```diff
@@ -98,6 +102,25 @@
...
+
+    & when (@config-colored-header = true) {
+      .Button--color(@control-color, @control-bg);
+
+      &:hover {
+        color: @link-color;
+      }
+    }
+
+    i.icon {
```
Comment (franzliedke) [inline less/forum/NotificationList.less]: I guess you dont' need the element selector here, just the class is enough and leaves our selector specificity more consistent.

File: js/src/forum/components/Notification.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Component from '../../common/Component';
 import avatar from '../../common/helpers/avatar';
 import icon from '../../common/helpers/icon';
 import humanTime from '../../common/helpers/humanTime';
+import Button from "../../common/components/Button";
```
Comment (franzliedke) [inline js/src/forum/components/Notification.js]: Quotes :scream:

Comment (luceos) [reply js/src/forum/components/Notification.js]: I guess it's time to add linting to ci 🙈

**Comment by datitisev** [comment]
🙈 I'll fix this in a bit, forgot about this.

**Comment by datitisev** [comment]
See #1874, I deleted my fork and it wouldn't continue the PR after I re-created it 🙈

**Pull request closed without merging**